### PR TITLE
feat(auth): realized simple route guard to private routes

### PR DESCRIPTION
**Pull request by DermanovskiyI**
implemented autorization state via redux-toolkit. Permission to follow guarded route is provided through the useCheckAuth hook

### CODE REVIEW

File: src/pages/home/Home.tsx [hunk 1]
```diff
@@ -1,25 +1,34 @@
...
+import { RootState } from '@/shared/model';
+import { useAppDispatch, useAppSelector } from '@/shared/use/redux';
+import { autorize } from '@/shared/model/auth-slice';
+const Home = () => {
+  const isLoginLoading = useAppSelector(
+    (state: RootState) => state.auth.isLoading
+  );
+  const dispatch = useAppDispatch();
+  const fetch = async () => {
+    await TodosService.fetchTodos();
```
Comment (no0dly) [inline src/pages/home/Home.tsx:12]: Looks like you dont need `async/await` here

File: src/pages/home/Home.tsx [hunk 2]
```diff
@@ -1,25 +1,34 @@
...
+  };
+  const handleAuth = async () => {
+    await dispatch(autorize());
```
Comment (no0dly) [inline src/pages/home/Home.tsx:15]: same here

File: src/pages/home/Home.tsx (hunk 1 above)
Comment (no0dly) [reply src/pages/home/Home.tsx:12]: Also function name `handleFetchTodos` fits better here

File: src/shared/layouts/PersonalAccLayout.tsx [hunk 3]
```diff
@@ -82,7 +89,12 @@ const PersonalAccLayout = () => {
           ))}
         </List>
         <Divider />
-        <Button variant="text">Выйти</Button>
+        <Button
+          variant="text"
+          onClick={() => dispatch(logout())}
```
Comment (no0dly) [inline src/shared/layouts/PersonalAccLayout.tsx:94]: Good practice to not make unnecessary anonymous functions  
```
const onClickHandle = () => dispatch(logout());

<Button
onClick={onClickHandle} />
```

File: src/shared/model/auth-slice/index.ts [hunk 4]
```diff
@@ -0,0 +1,58 @@
...
+  initialState,
+  reducers: {
+    logout: (state) => {
+      state.isAutorized = false;
+    },
+  },
+  extraReducers: (builder) => {
+    builder
+      .addCase(autorize.fulfilled, (state, action: PayloadAction<boolean>) => {
+        console.log('action', action.payload);
```
Comment (no0dly) [inline src/shared/model/auth-slice/index.ts:41]: Try this, should help you to log actions: https://github.com/LogRocket/redux-logger

**Review by no0dly — APPROVED** [review]
this is the ticket

**Comment by DermanovskiyI** [comment]
> this is the ticket

You listen. I speak. And it will be okey

**Pull request merged**